src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -593,9 +586,9 @@ private BoundLambda ReallyInferReturnType(NamedTypeSymbol delegateType, Immutabl
                 diagnostics: diagnostics);
             Binder lambdaBodyBinder = new ExecutableCodeBinder(_unboundLambda.Syntax, lambdaSymbol, ParameterBinder(lambdaSymbol, binder));
             var block = BindLambdaBody(lambdaSymbol, lambdaBodyBinder, diagnostics);
-            var returnTypes = ArrayBuilder<(RefKind, TypeSymbolWithAnnotations)>.GetInstance();
+            var returnTypes = ArrayBuilder<(BoundReturnStatement, TypeSymbolWithAnnotations)>.GetInstance();
             BoundLambda.BlockReturns.GetReturnTypes(returnTypes, block);
-            var inferredReturnType = BoundLambda.InferReturnType(returnTypes, lambdaBodyBinder.Compilation, lambdaBodyBinder.Conversions, delegateType, lambdaSymbol.IsAsync);
+            var inferredReturnType = BoundLambda.InferReturnType(returnTypes, block, lambdaBodyBinder.Compilation, lambdaBodyBinder.Conversions, delegateType, lambdaSymbol.IsAsync);
```

Comment:
>block [](start = 78, length = 5)

Why are we passing ```block``` instead of ```this``` here? What is the purpose of this parameter?

Reply:
`this` is an `UnboundLambdaState` which isn't a bound node.

---
In reply to: [244366298](https://github.com/dotnet/roslyn/pull/31813#discussion_r244366298) [](ancestors = 244366298)